lib/edgedb/edgeql/generator.ex

```diff
@@ -199,16 +204,19 @@ defmodule EdgeDB.EdgeQL.Generator do
       generate_query_module(
         query_file: query_file,
         module_name: module_name,
-        types: types(),
+        types: types,
         shape: rendered_shape,
         schema: rendered_schema,
-        query_function: @cardinality_to_function[query.result_cardinality],
-        should_render_type_for_shape: complex_shape?(raw_shape),
-        result_type: (complex_shape?(raw_shape) && "result()") || rendered_shape,
+        should_render_type_for_shape: rendered_schema && complex?,
+        cardinality_to_function: @cardinality_to_function,
```

Comment:
I think this is now useless, right? Since the template code checks the cardinality itself